### ISSUE TITLE
False positives with PEP 695 type aliases using Literal[SomeEnum.SomeValue]

### ISSUE THREAD

**Issue by mdbenito** [astral-sh/ty#3446]
### Summary

`ty` reports false-positive `invalid-argument-type` and `invalid-assignment` diagnostics when a PEP 695 type alias combines enum-member `Literal[...]` values inside a tuple union.

The same alias works if written with `typing.TypeAlias`.

The expected behavior is that there should be no diagnostics with the code below.

## Reproducer

```python
from enum import Enum
from typing import Final, Literal, TypeAlias


class AnySentinel(Enum):
    AnyNode = "AnyNode"
    AnyEdge = "AnyEdge"


AnyNode: Final = AnySentinel.AnyNode
AnyEdge: Final = AnySentinel.AnyEdge

type AnyNodeT = Literal[AnySentinel.AnyNode]  # If string literal, no false positive
type AnyEdgeT = Literal[AnySentinel.AnyEdge]  # If string literal, no false positive


type StoreSelector = AnyNodeT | AnyEdgeT


class FeatureKey:
    entity_t: StoreSelector


def needs_selector(selector: StoreSelector):
    pass


def accepts_optional_selector(selector: StoreSelector | None):
    pass


def check_feature_key(key: FeatureKey):
    # False positive: `entity_t` is declared as `StoreSelector`, but ty reports
    # the expanded union as including plain `AnySentinel`.
    needs_selector(key.entity_t)
    accepts_optional_selector(key.entity_t)


def check_annotated_empty_containers():
    # False positive: RHS is an empty container and LHS is explicitly annotated.
    selectors: list[StoreSelector] = []
    selector_map: dict[StoreSelector, bool] = {}

    # These work
    selectors.append(AnyNode)
    selector_map[AnyEdge] = True
```

## Actual Behavior

With ty 0.0.34 and ty 0.0.35, ty check reports:

```
error[invalid-argument-type]: Argument to function `needs_selector` is incorrect
Expected `StoreSelector`, found `type[Node] | AnySentinel | tuple[NodeT | AnyNodeT, AnyEdgeT, NodeT | AnyNodeT]`
 ```

It also reports errors for explicitly annotated empty containers:

```python
selectors: list[StoreSelector] = []
selector_map: dict[StoreSelector, bool] = {}
```

An LLM thinks that the key issue could be that ty expands Literal[AnySentinel.AnyNode] | Literal[AnySentinel.AnyEdge] into plain AnySentinel, then rejects that widened type as not assignable back to the alias that produced it.

## Workaround

The equivalent `typing.TypeAlias` syntax passes:

```python
AnyNodeTAlias: TypeAlias = Literal[AnySentinel.AnyNode]
AnyEdgeTAlias: TypeAlias = Literal[AnySentinel.AnyEdge]
ConnectionDefTAlias: TypeAlias = tuple[AnyNodeTAlias, AnyEdgeTAlias, AnyNodeTAlias]
StoreSelectorAlias: TypeAlias = AnyNodeTAlias | AnyEdgeTAlias | ConnectionDefTAlias

```

So the bug appears specific to PEP 695 type aliases.

### Version

0.0.34, 0.0.35

**Comment by sharkdp** [comment]
Thank you for reporting this. Something seems seriously broken here. It looks like we don't expand one type alias layer somehow.

Here's a smaller example that's potentially easier to read
```py
from enum import Enum
from typing import Literal

class AB(Enum):
    A = "A"
    B = "B"

type LiteralA = Literal[AB.A]
type LiteralB = Literal[AB.B]

def _(ab: AB):
    x: LiteralA | LiteralB = ab. # type `AB` is not assignable to `LiteralA | LiteralB`
```
